File: src/utils/logger.js
/**
 * Simple logger implementation for the Canvas-Discourse integration
 */
function createLogger(moduleName) {
  const loggerPrefix = `[${moduleName}]`;
  
  return {
    /**
     * Log an info message
     * @param {string} message - Message to log
     * @param {Object} [context={}] - Optional context data
     */
    info(message, context = {}) {
      console.log(`[INFO] ${loggerPrefix} ${message}`, context);
    },
    
    /**
     * Log an error message
     * @param {string} message - Message to log
     * @param {Object} [context={}] - Optional context data
     */
    error(message, context = {}) {
      console.error(`[ERROR] ${loggerPrefix} ${message}`, context);
    },
    
    /**
     * Log a warning message
     * @param {string} message - Message to log
     * @param {Object} [context={}] - Optional context data
     */
    warn(message, context = {}) {
      console.warn(`[WARN] ${loggerPrefix} ${message}`, context);
    },
    
    /**
     * Log a debug message
     * @param {string} message - Message to log
     * @param {Object} [context={}] - Optional context data
     */
    debug(message, context = {}) {
      if (process.env.DEBUG) {
        console.debug(`[DEBUG] ${loggerPrefix} ${message}`, context);
      }
    }
  };
}

module.exports = { createLogger };

File: src/utils/logger.rs
// src/utils/logger.rs
use log::{info, error, warn};
use serde_json::Value;
use std::fmt;

/// Simple logger implementation for the Canvas-Discourse integration
pub struct Logger {
    module_name: String,
}

impl Logger {
    /// Create a new logger instance for a specific module
    pub fn new(module_name: &str) -> Self {
        Logger {
            module_name: module_name.to_string(),
        }
    }

    /// Log an info message
    pub fn info<T: fmt::Display>(&self, message: T, context: Option<&Value>) {
        match context {
            Some(ctx) => info!("[{}] {} {:?}", self.module_name, message, ctx),
            None => info!("[{}] {}", self.module_name, message),
        }
    }

    /// Log an error message
    pub fn error<T: fmt::Display>(&self, message: T, context: Option<&Value>) {
        match context {
            Some(ctx) => error!("[{}] {} {:?}", self.module_name, message, ctx),
            None => error!("[{}] {}", self.module_name, message),
        }
    }

    /// Log a warning message
    pub fn warn<T: fmt::Display>(&self, message: T, context: Option<&Value>) {
        match context {
            Some(ctx) => warn!("[{}] {} {:?}", self.module_name, message, ctx),
            None => warn!("[{}] {}", self.module_name, message),
        }
    }
}

/// Create a logger instance for a specific module
pub fn create_logger(module_name: &str) -> Logger {
    Logger::new(module_name)
}
